core/vm/memory.go

```diff
@@ -85,8 +85,12 @@ func (m *Memory) Get(offset, size int64) (cpy []byte) {
 	return
 }
 
+// ReadOnlyMem represents a slice of memory which has not been copied,
+// but is a pointer directly to memory, and should not be modified
+type ReadOnlyMem []byte
```

Comment:
I'm not really a fan of this. `[]byte` doesn't have any methods itself, so aliasing won't hide anything. Calling code will still have access to all the slice functions and indexing and modification. I think this name just gives a false sense of security or confusion.

Reply:
removed